File: scripts/agent/convert_dds.py
"""DDS to PNG/JPEG converter using Pillow."""
import sys
from PIL import Image

if len(sys.argv) < 4:
    print("Usage: python convert_dds.py <input.dds> <output.png|jpg> <format: PNG|JPEG> [minimap]")
    sys.exit(1)

input_path = sys.argv[1]
output_path = sys.argv[2]
fmt = sys.argv[3].upper()
is_minimap = len(sys.argv) > 4 and sys.argv[4] == "minimap"

img = Image.open(input_path)
if fmt == "JPEG":
    img = img.convert("RGB")
else:
    img = img.convert("RGBA")

# Minimap: Immer die Hälfte der Pixel von der Mitte, dann auf 2048x2048 skalieren
if is_minimap:
    w, h = img.size
    crop_size = w // 2
    cx, cy = w // 2, h // 2
    half = crop_size // 2
    img = img.crop((cx - half, cy - half, cx + half, cy + half))
    if img.size[0] != 2048:
        img = img.resize((2048, 2048), Image.LANCZOS)

# Für HUD-Icons: auf 64x64 skalieren
elif "hud" in input_path.lower() or img.size[0] <= 256:
    img = img.resize((64, 64), Image.LANCZOS)

img.save(output_path, fmt, quality=85)
print(f"OK: {output_path} ({img.size[0]}x{img.size[1]})")

File: scripts/agent/convert-dds.ts
import { execSync } from "child_process";
import { existsSync } from "fs";
import path from "path";

const PYTHON_SCRIPT = path.join(__dirname, "convert_dds.py");

export function convertDDS(inputPath: string, outputPath: string, format: "PNG" | "JPEG", minimap?: boolean): boolean {
  if (!existsSync(inputPath)) {
    console.warn(`  ⚠ DDS nicht gefunden: ${inputPath}`);
    return false;
  }

  try {
    const minimapArg = minimap ? " minimap" : "";
    execSync(`python "${PYTHON_SCRIPT}" "${inputPath}" "${outputPath}" ${format}${minimapArg}`, {
      stdio: "pipe",
    });
    return true;
  } catch (error) {
    console.warn(`  ⚠ Konvertierung fehlgeschlagen: ${inputPath}`);
    return false;
  }
}
